donation-api/src/main/java/org/interfaithsanctuary/donationapi/model/Donation.java

```diff
@@ -0,0 +1,71 @@
+package org.interfaithsanctuary.donationapi.model;
+
+import javax.persistence.*;
+import java.util.Date;
+
+/**
+ * This object describes the many-many relationship that a Donation can have with a Callout.
+ */
+@Entity
+@Table(name = "donations")
+public class Donation {
+
+    @Id
+    @Column(name = "donation_id")
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name="calloutneed_id")
+    private CalloutNeed calloutNeed;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name="donor_id")
+    private Donor donor;
+
+    @Column(name = "donation_date")
+    private Date date;
+
+    @Column(name = "donation_quantity")
+    private int quantity;

```

Comment:
`quantity` should be an `Integer` because an `int` has a default value of zero, which could cause problems if persisted incorrectly. My understanding is that we want empty fields to be persisted as `null`.

Reply:
We have this setup as NOT NULL in the database so we can't enter a null value into the database. If they changed that in the future hopefully they will also update this field

Reply:
@TylerNicholls I was thinking persisting it as null can be a catch-all so that we don't get incorrect data in the database, like if we somehow still persist a Donation without a quantity (I'm assuming our front-end will check for a quantity) we'll get a server error back, rather than persisting an incorrect quantity of 0.

Reply:
@Tiernebre has done more tutorials on this stuff than me though so he might be able to provide more insight

Reply:
I never even thought of this to be honest. Really good idea. Thanks for the feedback I will change this too!

Reply:
Thanks man! When Brendan sees these comments he might be able to give more detailed feedback too :)